code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -191,12 +193,45 @@ void setNetwork(Map<String, Object> inspect, Instance instance) {
         Network network = networkDao.getNetworkForObject(instance, networkKind);
 
         if (network != null) {
+            if (!KIND_HOSTONLY.equals(networkKind)) {
+                Network rancherNetwork = networkDao.getNetworkForObject(instance, KIND_HOSTONLY);
+                String ip = getDockerIp(inspect);
+                if (StringUtils.isNotEmpty(ip) && isIpInNetwork(ip, rancherNetwork)) {
+                    DataAccessor.fields(instance).withKey(FIELD_REQUESTED_IP_ADDRESS).set(ip);
+                    network = rancherNetwork;
+                }
+            }
+
             List<Long> netIds = new ArrayList<Long>();
             netIds.add(network.getId());
             DataAccessor.fields(instance).withKey(FIELD_NETWORK_IDS).set(netIds);
         }
     }
 
+    boolean isIpInNetwork(String ipAddress, Network network) {
+        for (Subnet subnet : objectManager.children(network, Subnet.class)) {
+            String startIp = NetUtils.getDefaultStartAddress(subnet.getNetworkAddress(), subnet.getCidrSize());
```

Comment:
children() doesn't check for removed.isNull().  Can you check here.  This will shortly cause an issue.  We will add the support to change the network subnet which will mean we will remove the old one and add a new one.